eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubClient.java

```diff
@@ -149,41 +155,78 @@ public EventSender createSender(EventSenderOptions options) {
-     * @param options Additional options for the receiver.
-     * @return An new {@link EventReceiver} that receives events from the partition with all configured {@link EventReceiverOptions}.
-     * @throws NullPointerException if {@code partitionId}, {@code eventPosition}, or {@code options} is {@code null}.
+     * <p>
+     * A consumer may be exclusive, which asserts ownership over the partition for the consumer group to ensure that
+     * only one consumer from that group is reading the from the partition. These exclusive consumers are sometimes
+     * referred to as "Epoch Consumers."
+     *
+     * A consumer may also be non-exclusive, allowing multiple consumers from the same consumer group to be actively
+     * reading events from the partition. These non-exclusive consumers are sometimes referred to as "Non-epoch
+     * Consumers."
+     *
+     * Designating a consumer as exclusive may be specified in the {@code options}, by setting
+     * {@link EventHubConsumerOptions#ownerLevel(Long)} to a non-null value. By default, consumers are
+     * created as non-exclusive.
+     * </p>
+     *
+     * @param consumerGroup The name of the consumer group this consumer is associated with. Events are read in the
+     *         context of this group. The name of the consumer group that is created by default is
+     *         {@link #DEFAULT_CONSUMER_GROUP_NAME}.
+     * @param partitionId The identifier of the Event Hub partition from which events will be received.
+     * @param eventPosition The position within the partition where the consumer should begin reading events.
+     * @param options The set of options to apply when creating the consumer.
+     * @return An new {@link EventHubConsumer} that receives events from the partition with all configured
+     *         {@link EventHubConsumerOptions}.
+     * @throws NullPointerException If {@code eventPosition}, or {@code options} is {@code null}.
+     * @throws IllegalArgumentException If {@code consumerGroup} or {@code partitionId} is {@code null} or an empty
+     *         string.
      */
-    public EventReceiver createReceiver(String partitionId, EventPosition eventPosition, EventReceiverOptions options) {
-        Objects.requireNonNull(partitionId);
+    public EventHubConsumer createConsumer(String consumerGroup, String partitionId, EventPosition eventPosition,
+                                           EventHubConsumerOptions options) {
         Objects.requireNonNull(eventPosition);
         Objects.requireNonNull(options);
 
-        final EventReceiverOptions clonedOptions = options.clone();
+        if (ImplUtils.isNullOrEmpty(consumerGroup)) {
+            throw new IllegalArgumentException("'partitionId' cannot be null or empty.");
+        }
+        if (ImplUtils.isNullOrEmpty(consumerGroup)) {
+            throw new IllegalArgumentException("'partitionId' cannot be null or empty.");
+        }
```

Comment:
Copy/paste issues here? we are checking for ConsumerGroup in both `if` blocks and throwing error for partitionId

Reply:
Ahaha phew. Good catch. I definitely copy and pasted that. 😆